compiler/il/NodeExtension.hpp

```diff
@@ -58,8 +58,7 @@ class NodeExtension
     template <class T>
     T  setElem(uint16_t index, T elem)
        {
-       (T)(_data[index] = (uintptr_t)elem);
-       return (T)_data[index];
+       return (T)(_data[index] = (uintptr_t)elem);
```

Comment:
I think the compiler warns that `(T)(_data[index] = (uintptr_t)elem);` is an "unused expression" (or something along those lines). Basically, since the resulting value of the expression `_data[index] = (uintptr_t)elem` isn't used, there's no need to cast it to type `T`.

I *think* an equivalent fix would be:

```
_data[index] = (uintptr_t)elem;
return (T)(_data[index]);
```
or even

```
_data[index] = (uintptr_t)elem;
return elem;
```

I have to admit I generally really don't like type casts involving template arguments.